hal/network/ncp_client/quectel/quectel_ncp_client.cpp

```diff
@@ -1992,7 +2001,7 @@ int QuectelNcpClient::modemSoftPowerOff() {
 int QuectelNcpClient::modemHardReset(bool powerOff) {
     LOG(TRACE, "Hard resetting the modem");
 
-    if (ncpId() != PLATFORM_NCP_QUECTEL_BG95_M1 && ncpId() != PLATFORM_NCP_QUECTEL_BG95_MF) {
+    if (!isQuecLPWADevice()) {
```

Comment:
We need to be careful here and make sure that this works correctly. BG77 might fall into the other category.